src/server/game/Chat/Chat.cpp

```diff
@@ -168,75 +168,46 @@ bool ChatHandler::hasStringAbbr(char const* name, char const* part)
 
 void ChatHandler::SendSysMessage(const char *str, bool escapeCharacters)
 {
-    WorldPacket data;
-
-    // need copy to prevent corruption by strtok call in LineFromMessage original string
-    char* buf;
-    char* pos;
+    std::string msg{ str };
 
-    if (escapeCharacters && strchr(str, '|'))
+    // Replace every "|" with "||" in msg
+    if (escapeCharacters && msg.find('|') != std::string::npos)
     {
-        size_t startPos = 0;
-        std::ostringstream o;
-        while (char const* charPos = strchr(str + startPos, '|'))
-        {
-            o.write(str + startPos, charPos - str - startPos);
-            o << "||";
-            startPos = charPos - str + 1;
-        }
-        o.write(str + startPos, strlen(str) - startPos);
-        buf = strdup(o.str().c_str());
-    }
-    else
-    {
-        buf = strdup(str);
-    }
+        Tokenizer tokens{msg, '|'};
+        std::ostringstream stream;
+        for (size_t i = 0; i < tokens.size() - 1; ++i)
+            stream << tokens[i] << "||";
+        stream << tokens[tokens.size() - 1];
```

Comment:
Looks to me as if there should be braces + equal indent for the 3 `stream` lines in this `for` loop.

Reply:
I'm not sure what do you mean exactly, but do you want me to write this instead of for loop from 178 line?
```cpp
for(size_t i = 0; i < tokens.size() - 1; ++i) 
{
  stream << tokens[i];
  stream << "||";
}
```

Reply:
Jesus Christ no

Reply:
Then what?

Reply:
nothing, its fine

Reply:
Sorry for being less specific than usual. I thought it would look like this:
```cpp
        for (size_t i = 0; i < tokens.size() - 1; ++i)
        {
            stream << tokens[i] << "||";
            stream << tokens[tokens.size() - 1];
            msg = stream.str();
        }
```

Reply:
It would completely change semantics of those lines and those new semantics wouldn't make any sense.

Reply:
OK, thanks for verifying my suspicion. I am not a real coder, so I do make mistakes.